src/contracts/core/DelegationManager.sol

```diff
@@ -137,7 +145,7 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
      * @notice Callable only by the StrategyManager
      * @dev Should only ever be called in the event that the `staker` has no active deposits in EigenLayer.
      */
-    function undelegate(address staker) external onlyStrategyManager {
+    function undelegate(address staker) external onlyStrategyManager onlyNotFrozen(staker){
```

Comment:
why aren't we applying this at the StrategyManager level instead?

Reply:
yeah good point, fixed